### PR TITLE
Update examples for SS 0.4.0 and SR 0.3.0

### PR DESCRIPTION
**Pull request by EricGustin**
This PR updates the system example to be compatible with SmartSim 0.4.0 and to use new features from the most recent release.

### CODE REVIEW

File: theta/smartsim_theta_installer.sh [hunk 1]
```diff
@@ -23,5 +23,5 @@ make lib && pip install .
 cd ..
 cd smartsim
 conda install swig cmake git-lfs -y
-pip install .[ml]
-smart --device cpu --onnx
\ No newline at end of file
+pip install -e .[dev] keras==2.3 tensorflow==2.4.2 onnx==1.9.0
```
Comment (al-rigazzi) [inline theta/smartsim_theta_installer.sh:26]: Can we update this to just run `pip install .[ml]`?

File: thetagpu/README.md [hunk 2]
```diff
@@ -22,23 +22,23 @@ With this environment loaded, users will need to build and install both SmartSim
...
 after the correct conda environment was loaded.
 
 ```bash
 export CC=$(which gcc)
 export CXX=$(which g++)
-conda install swig cmake git-lfs 
+conda install swig cmake git-lfs
 conda install pytorch==1.7.1 torchvision==0.8.2 torchaudio==0.7.2 cudatoolkit=11.0 -c pytorch -y
-pip install . tensorflow==2.4.2 numpy==1.19.5 onnx==1.7 
+pip install . tensorflow==2.4.2 numpy==1.19.5 onnx==1.7
```
Comment (al-rigazzi) [inline thetagpu/README.md:35]: Here too, I think we can just use `pip install .[ml]`

**Review by al-rigazzi — APPROVED** [review]
Just a couple of comments, but I'm approving as it LGTM

File: theta/smartsim_theta_installer.sh (hunk 1 above)
Comment (EricGustin) [reply theta/smartsim_theta_installer.sh]: Adressed.

File: thetagpu/README.md (hunk 2 above)
Comment (EricGustin) [reply thetagpu/README.md]: Addressed.

**Pull request merged**